### ISSUE TITLE
False negative: redundant fn wrapper with set literal

### ISSUE THREAD

**Issue by jramosg** [clj-kondo/clj-kondo#2891]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

```text
clj-kondo v2026.05.25
```

**macro usage**

No.

**platform**

macOS, native arm64 binary installed with Homebrew.

**editor**

VS Code with Calva. The problem also occurs when running clj-kondo from the
terminal.

**problem**

The `:redundant-fn-wrapper` linter detects a redundant wrapper when a local
binding contains a set, but it misses the same wrapper when the code calls a
set literal.

Sets implement `IFn`. The first form in the repro can pass the set directly to
`map`, as shown in the second form.

**repro**

Run:

```shell
clj-kondo --lint - \
  --config '{:linters {:redundant-fn-wrapper {:level :warning}}}' <<'EOF'
(ns repro)

(map #(#{:a :b} %) [:a])

(map #{:a :b} [:a])

(let [s #{:a}] #(s %))
EOF
```

Output:

```text
<stdin>:7:16: warning: Redundant fn wrapper
linting took 46ms, errors: 0, warnings: 1
```

clj-kondo reports the local-binding wrapper on line 7 but does not report the
set-literal wrapper on line 3.

**config**

```clojure
{:linters {:redundant-fn-wrapper {:level :warning}}}
```

**expected behavior**

The linter should report `#(#{:a :b} %)` as a redundant function
wrapper. The equivalent form is:

```clojure
(map #{:a :b} [:a])
```

The warning for `(let [s #{:a}] #(s %))` on line 7 should remain.
